test/gtest/adam.hpp

```diff
@@ -258,9 +266,9 @@ struct AdamTest : public ::testing::TestWithParam<AdamTestCase>
         double threshold = std::numeric_limits<Tp>::epsilon();
         auto error       = miopen::rms_range(ref_param, param);
 
-        EXPECT_TRUE(miopen::range_distance(ref_param) == miopen::range_distance(param));
-        EXPECT_TRUE(error < threshold * 10) << "Error output beyond tolerance Error:" << error
-                                            << ",  Thresholdx10: " << threshold * 10;
+        ASSERT_EQ(miopen::range_distance(ref_param), miopen::range_distance(param));
```

Comment:
tsao phải đổi từ `EXPECT_TRUE()` thành `ASSERT_TRUE()` ở đây thế ạ? Từ giờ mình phải dùng assert thay vì expect ạ?

Reply:
Because this is a check for critical errors, if this check is failed, the following check are meaningless.

Reply:
i got it. But is this usage a "must" or just a "nice-to-have"? `EXPECT_TRUE` still seems to be ok to use, right?